src/utils/appData.spec.ts

```diff
@@ -0,0 +1,54 @@
+import {validateAppDataDocument } from './appData'
+
+const VALID_RESULT = { 
+  result: true,
+}
+
+test('Valid minimal document', async () => {
+  const validation = await validateAppDataDocument({
+    "version": "0.1.0",
+    "metadata": {}
+  })
+  expect(validation).toEqual(VALID_RESULT);
+});
+
+test('Valid minimal document + appCode', async () => {
+  const validation = await validateAppDataDocument({
+    "version": "0.1.0",
+    "appCode": "MyApp",
+    "metadata": {}
+  })
+  expect(validation).toEqual(VALID_RESULT);
+});
+
+
+test('Valid minimal document + appCode + referrer', async () => {
+  const validation = await validateAppDataDocument({
+    "version": "0.1.0",
+    "appCode": "MyApp",
+    "metadata": {
+      "referrer": {
+        "version": "0.1.0",
+        "address": "0xFEB4acf3df3cDEA7399794D0869ef76A6EfAff52"
+      }
+    }
+  })
+  expect(validation).toEqual(VALID_RESULT);
+});
+
+test('Invalid: No version', async () => {
+  const validation = await validateAppDataDocument({
+    "appCode": "MyApp",
+    "metadata": {}
+  })
+  expect(validation.result).toBeFalsy();
```

Comment:
Should we validate also that the error matches what we expect?

Reply:
Actually for some reason i just check and it doesn't return anything (returns undefined). Anyways, not too important i think. I can change the signature to return just the boolean or we should review more that lib...

I did it very shallowly the read of the lib